### DISCUSSION TITLE
Migrations have no effect on foreign keys (UUID)

### DISCUSSION THREAD

**Discussion by andre-sebastian** [adonisjs/core#1758] [Help]
Hello community, I have a problem with the references of the foreign keys and with the migrations.
When i run the command: `adonis migration:run `
This error appears: `RequestError: Could not create constraint or index.`

The migration status is this:
![image](https://user-images.githubusercontent.com/37244658/96326397-d8954f00-0ff5-11eb-9a04-d1b4bd7dffba.png)

### The tables that were not migrated are:
**1602286642983_incident_crime_schema** 
```
class IncidentCrimeSchema extends Schema {
  up () {
    this.create("incident_crimes", (table) => {
      table.uuid("incidentCrimeId").notNullable().primary()
      table.uuid("incidentId").notNullable().references("incidentId").inTable("incidents")
      table.uuid("crimeCategoryId").notNullable().references("crimeCategoryId").inTable("crime_categories")
    })
  }

  down () {
    this.drop('incident_crimes')
  }
}

module.exports = IncidentCrimeSchema

```

**1602286660162_incident_schema** 
```
class IncidentSchema extends Schema {
  up() {
    this.create("incidents", (table) => {
      table.uuid("incidentId").notNullable().primary();
      table.uuid("locationId").notNullable().references("locationId").inTable("locations")
      table.uuid("addressTypeId").notNullable().references("addressTypeId").inTable("address_types")
      table.datetime("reportDate").notNullable();
      table.string("reportTile",255).notNullable();
      table.string("reportDescription",500).notNullable();
      table.string("otherDetails",700).nullable();
      table.string("latitude",30).notNullable();
      table.string("longitude",30).notNullable();
      table.string("state",100).notNullable();
      table.timestamps();
    });
  }

  down() {
    this.drop("incidents");
  }
}

module.exports = IncidentSchema;
```
**1602286675378_person_in_incident_schema**
```
class PersonInIncidentSchema extends Schema {
  up() {
    this.create("person_in_incidents", (table) => {
      //table.increments()
      table.uuid("personInIncidentId").notNullable().primary();
      table.uuid("incidentId").notNullable().references("incidentId").inTable("incidents")
      table.uuid("personId").notNullable().references("personId").inTable("people")
    });
  }

  down() {
    this.drop("person_in_incidents");
  }
}

module.exports = PersonInIncidentSchema;

```
The database should look like this:
![image](https://user-images.githubusercontent.com/37244658/96326562-50b04480-0ff7-11eb-8567-312296e77410.png)

**Comment by SchillingPilot** [comment]
I know this is an older post, but maybe this answer will help someone.

When you get a cryptic error like `RequestError: Could not create constraint or index`, there's **_usually_** one of two causes (sometimes both). 

1. Check that  the foreign key is the **exact same** type as the key it maps to in the other schema.
2. Check the firing order of your migrations. Remember, they go from the top down and order matters.

In this case, I think all the types are UUID, so probably no problem there. However, `IncidentCrimeSchema` runs before `IncidentSchema` but it needs the `incidents` table to be created first so it can assign the constraint.

Cheers.
